tools/pipelines/templates/include-install-pnpm.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# include-install-pnpm
+#
+# This template can be included in pipelines to install pnpm with store caching enabled.
+
+parameters:
+# The path containing the project(s) to build.
+- name: buildDirectory
+  type: string
+
+# If set to true, the pnpm store will not be cached or restored from cache.
+- name: disableCache
+  type: boolean
+  default: false
+
+# The path to the pnpm store. The contents here will be cached and restored when using pnpm in a pipeline.
+- name: pnpmStorePath
+  type: string
+  default: $(Pipeline.Workspace)/.pnpm-store
+
+steps:
+- ${{ if eq(parameters.disableCache, false) }}:
+  - task: Cache@2
+    displayName: Cache pnpm store
+    timeoutInMinutes: 5
+    inputs:
+      # Caches are already scoped to individual pipelines, so no need to include the release group name or tag
+      # in the cache key
+      key: 'pnpm-store | "$(Agent.OS)" | ${{ parameters.buildDirectory }}/pnpm-lock.yaml'
+      path: ${{ parameters.pnpmStorePath }}
+      restoreKeys: |
+        pnpm-store | "$(Agent.OS)"
+
+- task: Bash@3
+  displayName: Install and configure pnpm
+  inputs:
+    targetType: 'inline'
+    workingDirectory: ${{ parameters.buildDirectory }}
+    condition: or(succeeded(), canceled())
```

Comment:
Nit: Might be worth a brief comment explaining why we want to continue on canceled here.